### ISSUE TITLE
Feature Request: Outline Customization

### ISSUE THREAD

**Issue by pltrant** [microsoft/vscode#52202]
Is it going to be possible for extensions to contribute `configurationDefaults` for the Outline (to show and to set follow cursor and sort type)? It would seem logical since we can do that for things like the minimap. It's beneficial for creating a curated experience with certain specialized extensions (and users can always override said defaults).

**Comment by noraj** [comment]
Detailed use cases making use of outline customization https://github.com/microsoft/vscode/issues/89378

**Comment by bemygreat** [comment]
Hope to implement this feature as soon as possible, I think it is quite useful.

**Comment by jrieken** [comment]
related to https://github.com/microsoft/vscode/issues/53262

**Comment by aderchox** [comment]
Unfortunately, neither Outline in the sidebar nor the Ctrl+Shift+O+: seem to be able to filter the symbols by their visibilities(e.g., only the public ones). There are also issues that are closed because they were said to be duplicates, while I couldn't find the main issue. Could anyone point me to the main issue so that I can support it?

**Comment by kumarharsh** [comment]
+1 to @aderchox's comment - several issues have been marked as duplicates for exposing visibility information in the outline. I have spent upwards of 1 hour on this, but can't find the canonical issue for this (maybe all outline+visibility issues got duped in a circular loop?). Seems like many of such issues (#49995, #61239) were marked as dupe of #23927 - which seems to have been implemented already but the outline still doesn't expose visibility information - so what gives?

And it's not just me - it's the top search result for `vscode outline private method`: [link](https://www.reddit.com/r/vscode/comments/gv2tye/show_privateprotectedpublic_in_outline_panel/)

@jrieken I realise your time is very important but kindly have a look when you have some.
